src/hub/tasks.py

```diff
@@ -0,0 +1,31 @@
+from datetime import timedelta
+
+from celery.decorators import periodic_task
+from celery.task.schedules import crontab
+from django.db.models import Count, F, Sum
+from django.db.models.functions import Coalesce
+
+from hub.models import Hub
+from researchhub.celery import QUEUE_HUBS
+
+
+@periodic_task(
+    run_every=crontab(minute=0, hour=0),
+    priority=5,
+    queue=QUEUE_HUBS,
+)
+def calculate_and_set_hub_counts():
```

Comment:
I had an idea when looking at this script. I'm not sure if this works, but it can be a potential optimization:

```python
    hubs = Hubs.objects.annotate(
        total_paper_count=Count("related_documents"),
        posts_discussion_count=Sum("related_documents__posts__discussion_count"),
        paper_discussion_count=Sum("related_documents__paper__discussion_count"),
    ).update(
        paper_count=F("total_paper_count"),
        discussion_count=F("posts_discussion_count") + F("paper_discussion_count")
    )
```

This might allow us to skip over the iterator and have everything done in SQL

Reply:
It's a good idea. Seems to not work for me. I'll have to debug this further